### PR TITLE
Disable simulator on mobile through css

### LINKED ISSUE #784: CircuitVerse Doesn't Work on Mobile Devices
**Describe the bug**
On mobile devices, you can't use the simulator!

**To Reproduce**
1. Go to CircuitVerse on a mobile device
2. Go to the simulator
3. Try to use it
4. You'll have a hard time using it

OR

1. Go to CircuitVerse on your browser
2. Inspect Element
3. Change Size to be similar to that of a mobile device
4. Go to the simulator
5. Try to use it
6. You'll have a hard time using the simulator

**Expected behavior**
I expect that using the simulator on any device should be easy and effortless. However, so much as trying to drag an input onto the circuit is frustrating and incredibly difficult.

**Screenshots**
I can't really provide screenshots.

**Desktop (please complete the following information):**
 - OS: Android (But this issue should be across all mobile devices)
 - Browser: Chrome
 - Version: 79.0

**Additional context**
No additional context.

**Are you working on this issue? (Yes/No)**
No.

### PR DESCRIPTION
**Pull request by JoshHeng**
Fixes #784 
Extended form @MrBartusek PR #806 

#### Describe the changes you have made in this pr -
Mobile simulator is disabled through a css rule checking the minimum screen width

### Screenshots of the changes (If any) -

(Typo in `Welcome` has been fixed since GIF recorded)

### CODE REVIEW

**Comment by fluctix** [comment]
Can you confirm embedded works well ?

**Comment by JoshHeng** [comment]
@fluctix Embedding is generally fine however the transparent `Fullscreen` GUI is too large.
There are other more sigificant styling problems on the website on mobile though, such as the project page.

File: public/js/simulator-checker.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+}
+
+function checkScreenSize() {
+    if (document.documentElement.clientWidth < 700) {
+        return false;
+    }
+    return true;
+}
+function hasTouchScreen() {
```
[the remark this answers is not in the document]
Comment (JoshHeng) [reply public/js/simulator-checker.js]: At this point the function is very easy to read and understand. Reducing the cognitive complexity would just be moving each check into its own function, which could make it harder to understand and is unneeded

**Comment by JoshHeng** [comment]
Added checks to see if a mouse is connected so devices which only have a touchscreen will be disabled.

Disabled message:

**Comment by JoshHeng** [comment]
More unit testing should be added, but that isn't relevant for this PR (so coveralls shouldn't matter)

**Comment by MrBartusek** [comment]
There are failed tests which have been fixed on the current master can you rebase to check if everything passes now?

**Comment by JoshHeng** [comment]
Failing AppVeyor/CircleCI build isn't caused by these changes

**Comment by tachyons** [comment]
Rebase with master

**Comment by nitin10s** [comment]
@JoshHeng Could please correct the failing checks?

**Comment by JoshHeng** [comment]
If anyone can give me guidance on fixing the circleci errors I'm happy to make more changes, but I don't think that this PR has caused them.

**Comment by abstrekt** [comment]
@nitin10s what to do with this PR?

**Comment by MrBartusek** [comment]
@JoshHeng that was a problem with master branch at this time, please rebase

**Comment by abstrekt** [comment]
maybe keep this on hold, merging it might cause problem for testing the simulator for touch screens, mobile screens, @nitin10s?

**Comment by JoshHeng** [comment]
Hi, should I merge new changes and rebase or wait? @abstrekt

**Comment by abstrekt** [comment]
> Hi, should I merge new changes and rebase or wait? @abstrekt 

I think you should wait, because there will be an attempt to make the simulator mobile friendly, merging this will cause problem while testing for mobile friendly..

**Comment by abstrekt** [comment]
@nitin10s can give this a look.

**Comment by nitin10s** [comment]
@JoshHeng It will be great if we can add this to the message: "We are working to make the Simulator mobile-friendly.  Make sure to check back with us later. Thank you for your patience." Also, please update the screenshots. Thanks.

**Comment by JoshHeng** [comment]
ezgif-6-c2d44e0b0695-min

I've merged into master however builds still seem to be failing. I don't think this is caused by this PR but because of another issue in master.

**Comment by tachyons** [comment]
Problem is with app/assets/javascripts/application.js require_tree is loaded before require jquery and that makes your file failed to use jquery and show error

**Comment by JoshHeng** [comment]
@nitin10s @tachyons Sorry for inactivity - please review.
Regarding the failing checks, I've looked through them and they all seem to exist from master rather than my branch specifically

**Comment by JoshHeng** [comment]
If the large diff in app/views/simulator/edit.html.erb is a problem (I wrapped everything in a `<main>` tag & indented wrapped code), I can remove the indentation

**Comment by tachyons** [comment]
Thanks for the great work, I think we may have to freeze this PR for now as simulator redesign is coming with better touch support.

**Pull request closed without merging**